File: Desktop/EY/aitax-poc/apps/backend/app/models/document.py
from datetime import datetime
from enum import Enum
from typing import List, Optional

from sqlmodel import Field, SQLModel, Relationship


class DocumentType(str, Enum):
    """Document types"""
    PDF = "pdf"
    TXT = "txt"


class DocumentBase(SQLModel):
    """Base document model"""
    title: str
    description: Optional[str] = None
    file_path: str
    file_type: DocumentType
    file_size: int  # Size in bytes
    user_id: int = Field(foreign_key="user.id")


class Document(DocumentBase, table=True):
    """Document model for database"""
    id: Optional[int] = Field(default=None, primary_key=True)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
    
    # Relationships
    notes: List["Note"] = Relationship(back_populates="document")
    chat_messages: List["ChatMessage"] = Relationship(back_populates="document")


class DocumentCreate(DocumentBase):
    """Document creation model"""
    pass


class DocumentRead(DocumentBase):
    """Document read model"""
    id: int
    created_at: datetime


class DocumentUpdate(SQLModel):
    """Document update model"""
    title: Optional[str] = None
    description: Optional[str] = None

File: Desktop/EY/aitax-poc/apps/frontend/src/types/document.ts
export enum DocumentType {
  PDF = "pdf",
  TXT = "txt"
}

export interface DocumentBase {
  title: string;
  description?: string;
  file_path: string;
  file_type: DocumentType;
  file_size: number;
  user_id: number;
}

export interface Document extends DocumentBase {
  id: number;
  created_at: string;
  updated_at: string;
}

export interface DocumentUpload {
  title: string;
  description?: string;
  file: File;
}
